src/RoleObjects/SliderData.js

```diff
@@ -1,115 +1,190 @@
-    */
+   * Retrieves the amount to adjust the slider by for page up/down keystrokes
+   * @access public
+   * @returns {Number} amount to increment or decrement the slider value by when
+   * page up or down is pressed.  undefined when browser default behavior is used
+   * for those keystrokes.
+   */
+  get pageStep() {
+    return this._pageStep;
+  }
+
+  /**
+   * Sets the orientation of slider
+   * @access public
+   * @param {String} str - "horizontal" for a horizontal slider, "vertical" for a vertical slider
+   */
   set orientation(str) {
     this._reactProps['aria-orientation'] = str;
   }
 
   /**
-    * Retrieves the orientation of slider
-    * @access public
-    * @returns  {String} str "horizontal" for a horizontal slider, "vertical" for a vertical slider
-    */
+   * Retrieves the orientation of slider
+   * @access public
+   * @returns  {String} str "horizontal" for a horizontal slider, "vertical" for a vertical slider
+   */
   get orientation() {
     return this._reactProps['aria-orientation'];
   }
 
   /**
-  * Set slider value
-  * @access public
-  * @param {Number} val - slider current value
-  */
+   * Set slider value
+   * @access public
+   * @param {Number} val - slider current value
+   */
   set value(val) {
     this._reactProps.value = val;
   }
 
   /**
-  * get slider value
-  * @access public
-  * @returns {Number} -slider current value
-  */
+   * get slider value
+   * @access public
+   * @returns {Number} -slider current value
+   */
   get value() {
     return this._reactProps.value;
   }
 
   /**
-  * Set slider min value
-  * @access public
-  * @param {Number} val - slider min value
-  */
+   * Set slider min value
+   * @access public
+   * @param {Number} val - slider min value
+   */
   set min(val) {
     this._reactProps.min = val;
   }
 
   /**
-  * get slider min value
-  * @access public
-  * @returns {Number} -slider min value
-  */
+   * get slider min value
+   * @access public
+   * @returns {Number} -slider min value
+   */
   get min() {
     return this._reactProps.min;
   }
 
   /**
-  * Set slider max value
-  * @access public
-  * @param {Number} val - slider max value
-  */
+   * Set slider max value
+   * @access public
+   * @param {Number} val - slider max value
+   */
   set max(val) {
     this._reactProps.max = val;
   }
 
   /**
-  * get slider max value
-  * @access public
-  * @returns {Number} - slider max value
-  */
+   * get slider max value
+   * @access public
+   * @returns {Number} - slider max value
+   */
   get max() {
     return this._reactProps.max;
   }
+
+  /**
+   * Event listener for React change events
+   * @access private
+   * @param {SyntheticEvent} evt - React event
+   */
+  _onChange(evt) {
+    this.value = evt.target.value;
+    const event = new createjs.Event('valueChanged', false, false);
+    event.newValue = this.value;
+    this._displayObject.dispatchEvent(event);
+  }
+
+  /**
+   * @inheritdoc
+   */
+  _onKeyDown(evt) {
```

Comment:
This function along with the new setters/getters in this file is the main part of the changes for the new functionality.